plugin/web/extensions/sidebar/popups/buildingSearch/components/ResultPanel/ResultItem.tsx

```diff
@@ -5,38 +5,43 @@ import { Result } from "../../types";
 
 type Props = {
   item: Result;
-  selected: string[];
-  onSelect: (selected: string[]) => void;
+  selected: Result[];
+  hasBorderBottom: boolean;
+  onSelect: (selected: Result[]) => void;
 };
 
-const ResultItem: React.FC<Props> = ({ item, selected, onSelect }) => {
+const ResultItem: React.FC<Props> = ({ item, selected, hasBorderBottom, onSelect }) => {
   const onClick = useCallback(() => {
-    if (selected.includes(item.id)) {
-      onSelect(selected.filter(sid => sid !== item.id));
+    if (selected.find(s => s.gml_id === item.gml_id)) {
+      onSelect(selected.filter(s => s.gml_id !== item.gml_id));
     } else {
-      onSelect([item.id]);
+      onSelect([item]);
     }
   }, [onSelect, item, selected]);
   return (
-    <StyledResultItem onClick={onClick} active={selected.includes(item.id)}>
-      {item.id}
+    <StyledResultItem
+      onClick={onClick}
+      active={!!selected.find(s => s.gml_id === item.gml_id)}
```

Comment:
I guess if `selected` props is huge data, it's better to memorize it.

Reply:
Currently `selected` should contain only one item, but might need support multiple even a lot of items in the future. I've updated this part with memorize. Thank you!